### DISCUSSION TITLE
Javers: Transaction ID

### DISCUSSION THREAD

**Discussion by lutsenko87** [javers/javers#1346] [General]
I want to create a Rollback functionality that will based on Javers audit. So my case: I have to save a list of entities ("Person") and I have to rollback this action due to some issue. I need to mark this transaction in Javers to find all the changed entities and roll back them regarding their revision.

Javers supports rollback functionality by the certain record but I need to select and rollback the whole group of entities.

Does Javers have such functionality or a workaround? Or maybe any ideas on how can I implement it.

**Comment by bartoszwalacik** [comment]
there is no explicit rollback functionality in Javers, because JaversRepository is append only. All you can do is to use DB rollback. The idea is to use the same transaction for your application changes and for Javers changes. Javers Spring starters are configured to work in this way.
